### PR TITLE
Use the docker image from the GitHub registry

### LINKED ISSUE #28903: DockerHub webplatformtests org limited to 3 members
@stephenmcgruer has let me know:

> Was going through handover tasks and realized we have a bit of a tricky issue with DockerHub (which we use to host wpt.fyi and wpt container images). DockerHub at some point added a member limit of 3 members for free organizations, and we were grandfathered in with 6 members.
>
> The 6 members currently are:
>
> gsnedders
> harjgam
> hexcles
> lastontheboat
> smcgruer
> wptfyibuilder 
>
> I would suggest we go through https://www.docker.com/community/open-source/application and try to get an OSS override.

Unfortunately, Stephen didn't hear back on the application, so we (@web-platform-tests/wpt-core-team) need to figure something out here. Right now, it's not possible to remove enough members to make room for me or @past in case we need access.

As an alternative, we could look into using [GitHub's Docker registry](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-docker-registry), which could probably be published to using GitHub Actions. But a risk is that Taskcluster needs to download these images, and who knows if GitHub is further away on the network, less reliable, or something...

### PR DESCRIPTION
**Pull request by OrKoN**
- updates publishing instructions for Docker images
- uses the [new image tagged `1`](https://github.com/web-platform-tests/wpt/pkgs/container/wpt/232922932?tag=1) hosted at GitHub

Closes https://github.com/web-platform-tests/wpt/issues/28903
Implements https://github.com/web-platform-tests/rfcs/pull/198

### CODE REVIEW

File: tools/docker/README.md [hunk 1]
```diff
@@ -1,16 +1,15 @@
...
 other tasks on Taskcluster. When any of the files in this directory change, the
-images must be updated as well. Doing this requires you be part of the
-'webplatformtests' organization on Docker Hub; ping @foolip or @jpchase
-if you are not a member.
+images must be updated as well. Doing this requires you to have write
+permissions to the repository.
 
 The tag for a new docker image is of the form
-`webplatformtests/wpt:{current-version + 0.01}`
+`{current-version + 0.01}`
```
Comment (jgraham) [inline tools/docker/README.md:7]: This is now wrong again. Also I'd take this chance to just move to integer version numbers.

**Review by jgraham — CHANGES_REQUESTED** [review]
I think actually merging this probably requires an RFC in case anyone is depending on the images in some other workflow.

**Comment by jgraham** [comment]
> do we need the local scripts in tools/docker/frontend.py? or is it sufficient to use the GitHub Actions workflow?

Yes, that's extremely useful for using the docker image locally, particularly when experimenting with changes to the image itself.

**Comment by OrKoN** [comment]
I have updated the PR but publishing a new image tagged `1` failed because launchpad.net is under maintenance so `apt-add-repository -y ppa:deadsnakes/ppa` fails.

Screenshot 2024-06-20 at 18 26 58

Related: https://askubuntu.com/a/1398569/374747

**Comment by OrKoN** [comment]
I think this PR is ready for review. Apart from landing this PR, an admin should set permissions on the package as documented in the linked RFC https://github.com/web-platform-tests/rfcs/pull/198

File: tools/docker/frontend.py [hunk 2]
```diff
@@ -56,17 +56,13 @@ def read_image_name():
...
-    resp = requests.get("https://hub.docker.com/v2/repositories/%s/%s/tags/%s" %
-                        (org, repo, version))
-    if resp.status_code == 200:
-        return True
-    if resp.status_code == 404:
-        return False
-    resp.raise_for_status()
+def tag_exists(tag):
+    retcode = subprocess.call(["docker",
+                        "manifest",
```
Comment (jgraham) [inline tools/docker/frontend.py:61]: Indentation looks a bit off here.

**Review by past — APPROVED** [review]
Looks good to me!

**Review by jgraham — APPROVED** [review]
Minor nits, but I think in general I'm OK with this landing.

**Pull request merged**